templates/vue-demo-store/components/LanguageSwitcher.vue

```diff
@@ -0,0 +1,32 @@
+<script setup lang="ts">
+import { getLanguageName } from "@shopware-pwa/helpers-next";
+
+const { languages, changeLanguage, replaceToDevStorefront } =
+  useInternationalization();
+const { languageIdChain } = useSessionContext();
+
+const onChangeHandler = async (option: Event) => {
+  const data = await changeLanguage((option.target as HTMLSelectElement).value);
+
+  if (data.redirectUrl) {
+    window.location.replace(replaceToDevStorefront(data.redirectUrl));
```

Comment:
I'm not sure that any Vue route function will be faster than the native browser function.

Reply:
I believe it should, as it should cause re-render without the need of loading all the assets. I'm fine with this if it's quick enough though :)
Wanted to test that, but I've found that there is some bug.

1. I went to preview: https://frontends-demo-git-feat-gh-222-shopware-frontends.vercel.app/
2. Switched language to "Polish"
3. I landed on page https://demo-frontends.shopware.store/pl-PL
![image](https://github.com/shopware/frontends/assets/13100280/ac6dd776-88eb-40be-9892-f81ed6491e45)

Reply:
I would also add that `data.redirectUrl` contains only the domain (with or without prefix/suffix etc). but being on specific pretty url like `/Summer-T-Shirt` for polish store may have a path like `/Wakacyjny-T-Shirt`. The redirect for always a main page isn't enough here and we need to combine domain + current path (translated if exists) when switching the language. Sounds like a bigger task for separate issue, right?

Reply:
@mdanilowicz ok, so we need to be allowed to pass our exact context like the current page we are on, so far, the backend does not know much at this point :) If we can achieve that on the backend it's totally fine, otherwise we need to take care of that